### DISCUSSION TITLE
Hints shown without asking!

### DISCUSSION THREAD

**Discussion by dco67** [rstudio/learnr#775] [Q&A]
I am developping exercises using learnr, and got stuck on the following problem: I add 'hints' to an exercise, hoping to see a HINT button showing up, but instead the hint text is posted below the code box, without any request from the user... It is just displayed as if it was regular text. This is shown below:

![image](https://user-images.githubusercontent.com/57354492/228870585-94e29da2-44de-439e-b54f-b41011915637.png)

The code that produced this is shown below. I can't see where is the error I made...

`

```{r mediation1, exercise = TRUE}
model2 <- "
pub ~ genre + ancien
ancien ~ genre
cit ~ ancien + genre + pub
salaire ~ pub + genre + ancien + cit
"
```

<div id="mediation-hint-1">
**Indice:** La valeur d'un effet indirect est le produit de deux coefficients, qu'il faudra **nommer**.
</div>

<div id="mediation-hint-2">
**Indice:** Pour nommer un paramètre, utilisez l'opérateur **:=**.
</div>

<div id="mediation-hint-3">
**Indice:** Nommez les coefficients liant `pub` à `cit`, puis `cit` à `salaire`.
</div>

```{r mediation1-solution}
model2 <- '
pub ~ genre + ancien
ancien ~ genre
cit ~ ancien + genre + b1*pub
salaire ~ pub + genre + ancien + b2*cit
mediation:= b1*b2
'
```

`
Any help would be greatly appreciated! Thanks... 
PS: the code is not reproduced correctly here... It doesn't show the < div id="mediation-hint-1" > .. < /div > tags. But they are indeed in the code I am using...

**Comment by gadenbuie** [comment]
Unfortunately, numbered `<div>`-style hints aren't supported in learnr. Currently your options are:

1. Multiple code hints, written in chunks with the chunk labels following the format `<label>-hint-<number>`, or
2. A single HTML hint wrapped in a `<div id="<label>-hint">` element.
